chia/wallet/uncurried_puzzle.py

```diff
@@ -11,5 +13,5 @@ class UncurriedPuzzle:
     args: Program
 
 
-def uncurry_puzzle(puzzle: Program) -> UncurriedPuzzle:
+def uncurry_puzzle(puzzle: Union[Program, SerializedProgram]) -> UncurriedPuzzle:
```

Comment:
if `puzzle` is a `SerializedProgram`, `uncurry()` returns a tuple of `SerializedProgram`. The member of `UncurriedPuzzle` is `Program`, `Program`. This doesn't look right

Reply:
If the puzzle is a `SerializedProgram`, `uncurry()` returns a tuple of `Program` as well.

Reply:
not according to the type stub: https://github.com/Chia-Network/chia_rs/blob/main/wheel/python/chia_rs/chia_rs.pyi#L1976

Reply:
oh, wait. `ChiaProgram` is an alias for the chia-blockchain python implementation